examples/ca-csr-example.cpp

```diff
@@ -0,0 +1,220 @@
+                );
+
+    // These are all supported X509 cert details supported by mococrw
+    auto rootCertDetails = DistinguishedName::Builder{}
+            .commonName("ImATeapot")
+            .countryName("DE")
+            .organizationName("Linux AG")
+            .organizationalUnitName("Linux Support")
+            .pkcs9EmailAddress("support@example.com")
+            .localityName("oben")
+            .stateOrProvinceName("nebenan")
+            .serialNumber("08E36DD501941432358AFE8256BC6EFD")
+            .givenName("Hot Teapot")
+            .userId("1000")
+            .title("Teapots contain tea")
+            .build();
+
+    return CertificateAuthority::createRootCertificate(
+                rootEccKey,
+                rootCertDetails,
+                // serial number
+                0,
+                rootCaSelfSignParams);
+}
+
+X509Certificate singCertificateWithRootCa(const X509Certificate &signingCert,
+                                          const AsymmetricKeypair &signingKey,
+                                          const CertificateSigningRequest &csr)
+{
+    auto ca = getRootCa(signingCert, signingKey);
+    try {
+        return ca.signCSR(csr);
+    } catch (const openssl::OpenSSLException &e) {
+        std::cerr << "Error reading CSR in PEM format. Please validate your CSR." << std::endl;
+        std::cerr << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    } catch (const MoCOCrWException &e) {
+        std::cerr << "Failed to create certificate." << std::endl;
+        std::cerr << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    }
+}
+
+X509Certificate signCertificateWithIntermediateCa(X509Certificate signingCert,
+                                                  const AsymmetricKeypair &signingKey,
+                                                  const CertificateSigningRequest &csr)
+{
+
+    auto ca = getIntermediateCa(signingCert, signingKey);
+    try {
+        return ca.signCSR(csr);
+    } catch (const openssl::OpenSSLException &e) {
+        std::cerr << "Error reading CSR in PEM format. Please validate your CSR." << std::endl;
+        std::cerr << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    } catch (const MoCOCrWException &e) {
+        std::cerr << "Failed to create certificate." << std::endl;
+        std::cerr << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    }
+}
+
+X509Certificate createIntermediateCaCertificate(const AsymmetricKeypair &intermediateCaKey,
+                                                const AsymmetricKeypair &rootCaKey,
+                                                const X509Certificate &rootCert)
+{
+    auto intermediateCaCertDetails = DistinguishedName(
+                DistinguishedName::Builder{}
+                .organizationalUnitName("Car IT")
+                .organizationName("BMW")
+                .countryName("DE")
+                .commonName("BMW internal CA Certificate").build()
+                );
```

Comment:
```suggestion
    auto intermediateCaCertDetails =
                DistinguishedName::Builder{}
                .organizationalUnitName("Car IT")
                .organizationName("BMW")
                .countryName("DE")
                .commonName("BMW internal CA Certificate").build();
```

No need to invoke the `DistinguishedName` copy constructor here, from what I can see?